File: src/pers/gargantua/mine_sweep/Ground.java
package pers.gargantua.mine_sweep;

import javax.swing.*;

public final class Ground extends JButton {
    
    public int r;
    public int c;
    public boolean isMine = false;
    public int around = 0;
    
    public Ground(int r, int c) {
        this.r = r;
        this.c = c;
        
        setFont(MyStyle.FONTB);
        setBackground(MyStyle.B);
    }
    
    public void setMine() {
        setText("M");
        setBackground(MyStyle.M);
    }
    
    public void setFlag() {
        setText("P");
        setBackground(MyStyle.P);
    }
    
    public void setNor() {
        if (around == 0) {
            setBackground(MyStyle.ZERO);
        }
        setText(String.valueOf(around));
        setForeground(MyStyle.NUMBER.get(around));
    }
    
    public void setQue() {
        setText("?");
        setBackground(MyStyle.Q);
    }
    
    public void setNew() {
        setText("");
        setBackground(MyStyle.B);
    }
}

File: src/pers/gargantua/mine_sweep/Ground.kt
package pers.gargantua.mine_sweep

import javax.swing.JButton

/**
 * @author Gargantua丶
 **/
class Ground(val r: Int, val c: Int, var isMine: Boolean = false, var around: Int = 0) : JButton() {

    init {
        font = MyStyle.FONT_B
        background = MyStyle.B
    }

    fun setMine() {
        text = "M"
        background = MyStyle.M
    }

    fun setFlag() {
        text = "F"
        background = MyStyle.F
    }

    fun setNor() {
        if (around == 0)
            background = MyStyle.ZERO
        text = around.toString()
        foreground = MyStyle.NUMBER[around]
    }

    fun setQue() {
        text = "?"
        background = MyStyle.B
    }

    fun setNew() {
        text = ""
        background = MyStyle.B
    }
}